src/sentry/grouping/strategies/newstyle.py

```diff
@@ -276,6 +287,10 @@ def frame(frame, event, context, **meta):
     if context_line_component is not None:
         values.append(context_line_component)
 
+    if context["use_package_fallback"] and all(component.is_empty() for component in values):
+        package_component = get_package_component(package=frame.package, platform=platform)
+        values.append(package_component)
```

Comment:
You should always append the package component and just overwrite `contributes`.